Update to GNOME 42

hfiguiere opened the pull request:
Close #27 
Close #28 
Close #29

hfiguiere commented:
@aleb @thiblahute can I merge this ?

aleb commented:
I don't have time to test. I'll get back after we update the dev env.

thiblahute commented:
Our [developement environment](https://developer.pitivi.org/HACKING.html?gi-language=undefined) in Pitivi is based on Flatpak and we should always have the flatpak manifest from there in sync with this one, starting by updating the one in the pitivi repo and then merging updating this on.

hfiguiere commented:
The thing is that now 3.38 is deprecated and you never took the patch.

thiblahute commented:
@aleb this should be good to go now right?

aleb commented:
Not sure if there are issues with 40. The dev env is on 41 now.

aleb commented:
Then we need to change the pygobject path in the json manifest.

hfiguiere commented:
you already need to change the pre-built wheels because the python version got upgraded.

hfiguiere commented:
Reworked it to update to GNOME 41:

- updated python package for 3.9 compatibility
- updated gstreamer to latest for the branch
- patched appdata for missing screenshot

hfiguiere commented:
evidence that:
1. python package should be build from source
2. that I only built on aarch64

hfiguiere commented:
bot, build

hfiguiere commented:
This works here on x86_64

hfiguiere commented:
Now with GNOME 42.

hfiguiere commented:
build fail because libcanberra.

hfiguiere commented:
this update uses libcanberra from shared-modules

hfiguiere commented:
bot, build

hfiguiere commented:
bot, build

hfiguiere commented:
that build failure is back...

A6GibKm commented:
I think that gst-plugging-base is in the runtime. See the error I am getting in my PR, the metainfo also needs patching since the screenshots urls do not exist anymore.

hfiguiere commented:
> I think that gst-plugging-base is in the runtime. See the error I am getting in my PR, the metainfo also needs patching since the screenshots urls do not exist anymore.

Updated gstreamer and patched the appdata.

hfiguiere commented:
This build is broken because of gst-python: python path did change. Will update.

aleb commented on pitivi-appdata.patch:
```diff
@@ -0,0 +1,13 @@
+   <screenshots>
+     <screenshot type="default">
+-      <image>http://www.pitivi.org/i/screenshots/software_center_main.jpg</image>
++      <image>http://www.pitivi.org/i/screenshots/2020.05.png</image>
```
https

hfiguiere replied:
done

aleb commented:
Can we just copy `shared-modules/libcanberra/libcanberra.json` over instead of using the shared-modules submodule? I'd rather never have to deal with git submodules again.

hfiguiere commented:
> Can we just copy `shared-modules/libcanberra/libcanberra.json` over instead of using the shared-modules submodule? I'd rather never have to deal with git submodules again.

And then it end up back to the same situation where things break and don't get updated. `shared-modules` are maintained properly and save a lot of time.

A6GibKm commented:
> Can we just copy `shared-modules/libcanberra/libcanberra.json` over instead of using the shared-modules submodule? I'd rather never have to deal with git submodules again.

The point of using it is precisely so that modules are well maintained and not broken, note that this is precisely what happened here where it needed both a patch and a newer url.

aleb commented:
The submodule also has to be updated, it does not update by itself, no? 

It's great that the shared-modules is well maintained, I'm all for copying what we need from there.

Am I missing anything?

Anyway, I don't even have merge rights in this repo, so don't mind me

hfiguiere commented:
> The submodule also has to be updated, it does not update by itself, no?

Nothing updates by itself, but this is the path of least resistance.

> Anyway, I don't even have merge rights in this repo, so don't mind me. You seem to know what you are doing. 👍

Ok I'll merge then.

hfiguiere commented:
Also added fallback-x11 as per #29
